### PR TITLE
elastic bean stalk application implementation

### PR DESCRIPTION
**Pull request by hongil0316**
## Description

Fixes https://github.com/gruntwork-io/cloud-nuke/issues/351.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.
- [ ] Attention Grunts - if this PR adds support for a new resource, ensure the `nuke_sandbox` and `nuke_phxdevops` jobs in `.circleci/config.yml` have been updated with appropriate exclusions (either directly in the job or via the `.circleci/nuke_config.yml` file) to prevent nuking IAM roles, groups, resources, etc that are important for the test accounts.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

**Review by denis256 — APPROVED** [review]

File: aws/resources/elastic_beanstalk.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+		if err != nil {
+			logging.Debugf("[Failed] %s", err)
+			continue
+		}
+
+		// get the deleted ids
+		deletedApps = append(deletedApps, id)
+		logging.Debugf("Deleted Elastic Beanstalk application: %s", *id)
+
```
Comment (denis256) [inline aws/resources/elastic_beanstalk.go:67]:
```
aws/resources/elastic_beanstalk.go:67: unnecessary trailing newline (whitespace)

	}
```

File: aws/resources/elastic_beanstalk_test.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+			configObj: config.ResourceType{
+				ExcludeRule: config.FilterRule{
+					TimeBefore: aws.Time(now.Add(1)),
+				}},
+			expected: []string{app2},
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
```
Comment (denis256) [inline aws/resources/elastic_beanstalk_test.go:91]: Was wondering why not to use `t.Parallel()`

File: aws/resources/elastic_beanstalk.go (hunk 1 above)
Comment (hongil0316) [reply aws/resources/elastic_beanstalk.go:67]: will reformat later!

File: aws/resources/elastic_beanstalk_test.go (hunk 2 above)
Comment (hongil0316) [reply aws/resources/elastic_beanstalk_test.go:91]: We should be able to use that. All the tests are mocked so it's pretty fast so no need to parallelized for performance at this point but we can definitely enable parallelism on all.

**Pull request merged**